src/axom/inlet/Table.cpp

```diff
@@ -161,6 +188,60 @@ Table& Table::addGenericArray(const std::string& name,
   return table;
 }
 
+Verifiable& Table::addBoolDict(const std::string& name,
+                               const std::string& description)
+{
+  return addPrimitiveArray<bool>(name, description, true);
+}
+
+Verifiable& Table::addIntDict(const std::string& name,
+                              const std::string& description)
+{
+  return addPrimitiveArray<int>(name, description, true);
+}
+
+Verifiable& Table::addDoubleDict(const std::string& name,
+                                 const std::string& description)
+{
+  return addPrimitiveArray<double>(name, description, true);
+}
+
+Verifiable& Table::addStringDict(const std::string& name,
+                                 const std::string& description)
+{
+  return addPrimitiveArray<std::string>(name, description, true);
+}
+
+Table& Table::addGenericDict(const std::string& name,
+                             const std::string& description)
+{
+  if(isGenericContainer())
+  {
+    SLIC_ERROR(fmt::format(
+      "[Inlet] Adding dict of structs to container of structs {0} is "
+      "not supported",
+      m_name));
+  }
+  auto& table = addTable(appendPrefix(name, CONTAINER_GROUP_NAME), description);
+  std::vector<std::string> indices;
+  const std::string& fullName = appendPrefix(m_name, name);
+  if(m_reader.getIndices(fullName, indices))
+  {
+    auto group = table.m_sidreGroup->createGroup(CONTAINER_INDICES_NAME,
+                                                 /* list_format = */ true);
+    for(const auto idx : indices)
+    {
+      table.addTable(idx, description);
+      group->createViewString("", idx);
```

Comment:
Should probably add a comment here (and other similar functions) that says something to the effect of:

Creating groups with indices for their names to be filled by following add functions on returned table.